packages/jellyfish-api-core/__tests__/category/spv/refundHtlcAll.test.ts

```diff
@@ -0,0 +1,219 @@
+    testing = Testing.create(new MasterNodeRegTestContainer())
+    container = testing.container
+    await container.start()
+    await container.spv.fundAddress(await container.call('spv_getnewaddress')) // Funds 1 BTC
+  })
+
+  afterEach(async () => {
+    await container.stop()
+  })
+
+  it('should refundHtlcAll', async () => {
+    const pubKeyA = await container.call('spv_getaddresspubkey', [await container.call('spv_getnewaddress')])
+    const pubKeyB = await container.call('spv_getaddresspubkey', [await container.call('spv_getnewaddress')])
+    const htlc1 = await container.call('spv_createhtlc', [pubKeyA, pubKeyB, '10'])
+    const htlc2 = await container.call('spv_createhtlc', [pubKeyA, pubKeyB, '10'])
+
+    const fund1 = await container.call('spv_sendtoaddress', [htlc1.address, 0.1]) // Funds HTLC address
+    const fund2 = await container.call('spv_sendtoaddress', [htlc2.address, 0.2])
+    await container.spv.increaseSpvHeight(10)
+
+    const destinationAddress = await container.call('spv_getnewaddress')
+    const results = await testing.rpc.spv.refundHtlcAll(destinationAddress) // This refund should only happen after timeout threshold set in createHtlc. See https://en.bitcoin.it/wiki/BIP_0199
+    expect(results.length).toStrictEqual(1)
+    for (const txid of results) {
+      expect(typeof txid).toStrictEqual('string')
+      expect(txid.length).toStrictEqual(64)
+    }
+
+    /**
+     * Assert that refund happened
+     * confirms at 11 as it `increaseSpvHeight` increased height by 10,
+     * and refund action adds 1.
+     */
+    const outputList: any[] = await container.call('spv_listhtlcoutputs')
+    expect(outputList.length).toStrictEqual(2)
+    const output1 = outputList.find(({ txid }: { txid: string }) => txid === fund1.txid)
+    const output2 = outputList.find(({ txid }: { txid: string }) => txid === fund2.txid)
+    expect(output1).toStrictEqual({
+      txid: fund1.txid,
+      vout: expect.any(Number),
+      amount: new BigNumber(0.1).toNumber(),
+      address: htlc1.address,
+      confirms: 11,
+      spent: {
+        txid: results[0],
+        confirms: 1
+      }
+    })
+    expect(output2).toStrictEqual({
+      txid: fund2.txid,
+      vout: expect.any(Number),
+      amount: new BigNumber(0.2).toNumber(),
+      address: htlc2.address,
+      confirms: 11,
+      spent: {
+        txid: results[0],
+        confirms: 1
+      }
+    })
+
+    /**
+     * Assert that the destination address received the refund
+     */
+    const listReceivingAddresses: any[] = await container.call('spv_listreceivedbyaddress')
+    const receivingAddress = listReceivingAddresses.find(({ address }: { address: string }) => address === destinationAddress)
+    expect(receivingAddress.address).toStrictEqual(destinationAddress)
+    expect(receivingAddress.txids.some((txid: string) => txid === results[0])).toStrictEqual(true)
+  })
+
+  it('should refundHtlcAll for expired only', async () => {
```

Comment:
do you expect `htlc2` to be no refunded in this test? wondering where is the `for expired only`
Refer https://github.com/DeFiCh/jellyfish/pull/529#discussion_r688293825

Reply:
Yes I do. In real life scenario, `refundHtlc` would probably not be called before timeout. However in this case, `refundHtlcAll` would be called in a real life scenario, as it initiate refunds only for those that already timeout, and not htlc that has not timed out yet.

Reply:
Should I change the title to indicate for timeout?